bftengine/src/preprocessor/RequestProcessingState.cpp

```diff
@@ -103,7 +103,10 @@ void RequestProcessingState::handlePreProcessReplyMsg(const PreProcessReplyMsgSh
   if (preProcessReplyMsg->status() == STATUS_GOOD) {
     numOfReceivedReplies_++;
     const auto &newHashArray = convertToArray(preProcessReplyMsg->resultsHash());
-    preProcessingResultHashes_[newHashArray]++;  // Count equal hashes
+    // Counts equal hashes and saves the signatures with the replica ID. They will be used as a proof that the primary
+    // is sending correct preexecution result to the rest of the replicas.
+    preProcessingResultHashes_[newHashArray].emplace_back(preProcessReplyMsg->getResultHashSignature(),
+                                                          preProcessReplyMsg->senderId());
```

Comment:
Didn't you lose "++"?

Reply:
Sorry, but I don't follow. Which value should be incremented?

Reply:
Forget about it, it's OK.